manta-pay/abstract-protocol.tex

```diff
@@ -143,29 +116,42 @@ \subsection{Abstract Cryptographic Schemes}
             \[\prove^P_\pk(x, w) := \prove(P, \pk, x, w)\]
         \item We write the $\VerifyingKey$ argument of $\verify$ in the subscript,
             \[\verify_\vk(x, \pi) := \verify(\vk, x, \pi)\]
-        \item We say that $(x, w) : \PublicInput \times \SecretInput$ has the property of being a $\satisfying$ input whenever
-            \[\satisfying^P_\pk(x, w) := \exists \pi : \Proof,\, \Some(\pi) \in \prove^P_\pk(x, w)\]
+        \item Given $P \colon \Statement$ and $\pk \colon \ProvingKey$, we define the function
+        \begin{align*}
+            \satisfying^P_\pk : \PublicInput \times \SecretInput \longrightarrow \Bool, 
+        \end{align*}
+        which is true if $\exists \pi : \Proof$ such that $\Some(\pi) \in \prove^P_\pk(x, w)$ and false otherwise. If $\satisfying^P_\pk (x, w) = \True$, we call the pair $(x,w)$ a \emph{satisfying input}.
```

Comment:
My point is just that because a Groth proof has two random blinding factors `r, s` it's possible (albeit unlikely) to make proofs according to the `prove` algorithm for any pair `(x, w)`.

Reply:
Claim: Given `x, w` a proof that they are satisfying exists.  
Sketch:

1.  Sample `r`, let quotient polynomial `h(X) = 0`, do not yet choose `s`. 
2.  Regard `A`, `B`, `C` of proof as `A = const.`, `B = const. + s * const.`, `C = const. + s * const.` where `const` summarizes any terms of  `A`, `B`, `C` that don't involve `s`. (and `const.` denotes a different group element each time it appears)
3.  Observe that `verify` equation reduces to `const. = s * const.`
4.  Solve for discrete log `s`, fill in above.

So a proof exists for `x, w`, it's just not feasible to find it unless `x, w` are "satisfying" in the sense of a relation. But this means all pairs are "satisfying inputs" in the sense it's defined here, so maybe we want to modify this def.

Reply:
Yeah we should use the strong variant here.